r/src/array_to_vector.cpp

```diff
@@ -1068,8 +1080,9 @@ std::shared_ptr<Converter> Converter::Make(const std::shared_ptr<DataType>& type
       return std::make_shared<arrow::r::Converter_Timestamp<int64_t>>(std::move(arrays));
 
     case Type::INT64:
-      // Prefer integer if it fits
-      if (ArraysCanFitInteger(arrays)) {
+      // Prefer integer if it fits, unless option arrow.int64_auto_downcast is `false`
+      if (ArraysCanFitInteger(arrays) &&
+          GetBoolOption("arrow.int64_auto_downcast", true)) {
```

Comment:
Your other comment said you were going with `arrow.int64_downcast` but that's not what is here. I'm ok either way though `arrow.int64_downcast` is more concise (and the "auto" behavior is really about what the default value is).